src/backend/cdb/motion/test/cdbsenddummypacket_test.c

```diff
@@ -296,21 +296,46 @@ test_send_dummy_packet_ipv6_to_ipv6_wildcard(void **state)
 	wait_for_receiver(false);
 }
 
+
+/*
+ * Test if IPv6 is supported.
+ *
+ * Sometimes even if IPv6 socket is successfully created it is not possible
+ * to bind an address to it (if IPv6 is disabled).
+ * 
+ * Note: this is not a general solution and should only be used for this
+ * specific test. It is theoretically possible that loopback IPv6 address
+ * is available but real IPv6 addresses aren't. However, checking for that
+ * would be a lot more complicated
+ */
+static bool is_ipv6_supported(void) {
+	int sockfd = socket(AF_INET6, SOCK_DGRAM, 0);
+	if (sockfd < 0 && errno == EAFNOSUPPORT)
+		return false;
+
+	struct sockaddr_in6 socket_struct;
+	socket_struct.sin6_family = AF_INET6;
+	socket_struct.sin6_addr = in6addr_loopback;
+	socket_struct.sin6_port = 0;
+	socket_struct.sin6_scope_id = 0;
```

Comment:
Maybe shorten a little:
```suggestion
	const struct sockaddr_in6 socket_struct =
	{ 
		.sin6_family = AF_INET6,
		.sin6_addr = in6addr_loopback
	};
```

Reply:
or even more
```suggestion
struct sockaddr_in6 socket_struct = {.sin6_family = AF_INET6, .sin6_addr = in6addr_loopback};
```

Reply:
> or even more

Some netinet struct members are not actual struct members, but access macros instead. But I guess this is safe, since `man sockaddr_in6` shows these as struct members.